File: Medium/Python/274. H-Index.py
class Solution(object):
    def hIndex(self, citations):
        length = len(citations)
        my_list = [0] * (length + 1)

        for num in citations:
            if length <= num:
                my_list[length] += 1
            else:
                my_list[num] += 1
        
        count = 0
        for i in range(length, 0, -1):
            count += my_list[i]
            if i <= count:
                return i
        
        return 0

File: Medium/Java/274. H-Index.java
class Solution {
    public int hIndex(int[] citations) {
        int len = citations.length;
        int[] arr = new int[len + 1];

        for(int num : citations){
            if(len <= num)
                arr[len]++;
            else
                arr[num]++;
        }

        int count = 0;
        for(int i = len; 0 <= i; i--){
            count += arr[i];
            if(i <= count)
                return i;
        }

        return 0;
    }
}
